src/inventory.h

```diff
@@ -108,15 +108,19 @@ struct ItemStack
 	}
 
 	// Get tool digging properties, or those of the hand if not a tool
-	const ToolCapabilities& getToolCapabilities(
+	ToolCapabilities getToolCapabilities(
```

Comment:
@nerzhul I could give itemstack metas a shadow ToolCapabilities value that gets updated on deserialization and every set to the toolcaps key, would that be better?